src/Categories/Functor/CartesianClosed.agda

```diff
@@ -0,0 +1,37 @@
+{-# OPTIONS --without-K --safe #-}
+
+module Categories.Functor.CartesianClosed where
+
+open import Level
+
+open import Categories.Category.CartesianClosed.Bundle using (CartesianClosedCategory)
+open import Categories.Category.CartesianClosed
+open import Categories.Functor using (Functor; _∘F_)
+open import Categories.Functor.Cartesian
+
+import Categories.Morphism as M
+
+private
+  variable
+    o ℓ e o′ ℓ′ e′ : Level
+
+record IsCartesianClosedF (C : CartesianClosedCategory o ℓ e) (D : CartesianClosedCategory o′ ℓ′ e′)
+  (F : Functor (CartesianClosedCategory.U C) (CartesianClosedCategory.U D)) : Set (o ⊔ ℓ ⊔ e ⊔ o′ ⊔ ℓ′ ⊔ e′) where
+  private
+    module C = CartesianClosedCategory C
+    module D = CartesianClosedCategory D
+    open Functor F
+    module CC = CartesianClosed C.cartesianClosed
+    module DC = CartesianClosed D.cartesianClosed
+  field
+    F-cartesian : IsCartesianF C.cartesianCategory D.cartesianCategory F
+  private
```

Comment:
Why make `F-mor` private? I wouldn't be surprised if many applications ended up needing it.

Reply:
By the way, what should it be named? Some sort of canonical morphism?

Reply:
+1 for asking on Zulip, but I don't think these sort of canonical morphisms have standard names. In nlab [the corresponding morphism for closed functors](https://ncatlab.org/nlab/show/closed+functor#definition) is just denoted $\hat{F}$. And it seems similar with the maps in e.g. monoidal functors, which are just given some Greek names (not following any particular convention). In the library these are called `ε` and `⊗-homo`. So by analogy this would be `^-homo`. Maybe we could come up with our own conventions? E.g. if a morphism witnesses the (lax) preservation of X (here X = exponentials) it could be called `preserves-X`? (Not particularly inventive, I know).

Reply:
Maybe we can open a separate PR to properly name all these.

Reply:
Sure, that can be done as a second step.